src/resources/DocumentResource.ts

```diff
@@ -0,0 +1,40 @@
+import FormData from 'form-data';
+import { Readable } from 'node:stream';
+
+import { DocumentMetadata, UploadDocumentRequestPayload } from '../types';
+import { Resource } from './Resource';
+
+/**
+ * Resource class to interact with
+ * {@link https://community.onespan.com/documentation/onespan-sign/guides/feature-guides/developer/uploading-deleting-documents | documents}.
+ *
+ * @public
+ */
+export class DocumentResource extends Resource {
+  /**
+   * Uploads a document to an existing package.
+   *
+   * @param packageId - Package ID
+   * @param payload - Metadata of the to-be-uploaded document
+   * @param documentBody - Data of the document
+   * @returns Metadata of the uploaded document
+   *
+   * @remarks
+   * - {@link https://community.onespan.com/products/onespan-sign/sandbox#/Documents/api.packages._packageId.documents.post | REST API documentation (OneSpan)}
+   *
+   * - {@link https://community.onespan.com/documentation/onespan-sign/guides/feature-guides/developer/uploading-deleting-documents | Uploading & Deleting Documents (OneSpan)}
+   */
+  public async create(
+    packageId: string,
+    payload: UploadDocumentRequestPayload,
```

Comment:
This payload and Packages's seem to follow a different naming convention: `payload: Requests.PackageData` --> Why?

Reply:
I'm moving requests and response payloads into their own namespace in this PR, but I missed this type 😅 My bad